rust/lance/src/dataset.rs

```diff
@@ -1455,6 +1482,56 @@ impl Dataset {
         *self = self.checkout_version(latest_version).await?;
         Ok(())
     }
+
+    /// List commit messages for versions of the dataset.
+    ///
+    /// This function retrieves commit messages associated with dataset versions.
+    ///
+    /// # Parameters
+    ///
+    /// * `max_transactions` - Maximum number of transactions to return. This limits the number of
+    ///   versions processed, which can improve performance for datasets with many versions.
+    ///   If `None`, all transactions will be returned.
+    ///
+    /// # Returns
+    ///
+    /// Returns a vector of tuples containing version number and commit message (if available).
+    pub async fn list_commit_messages(
```

Comment:
I think it will be more useful to just have methods like `list_transactions`  and then you could build `list_commit_message` on that.

Reply:
> it seems not easy to extract that properties from python level, since it was already str and looks like a json.

I think the core problem is that it's returning the transaction as a string? (that's also probably why it's in debug module)

if we have a list_transactions function that returns proper `Transaction`, would that fix the issue?

And I think having `list_transactions` in either python-ffi or rust works, I was thinking `list_transactions` would be pretty handy so maybe worth having that in rust directly.